main.go

```diff
@@ -0,0 +1,196 @@
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// A Recipe Struct allows you to insert recipe documents into your
+// collection
+
+type Recipe struct {
+	Name              string
+	Ingredients       []string
+	PrepTimeInMinutes int `json:"prepTimeInMinutes" bson:"prepTimeInMinutes"`
+}
+
+func main() {
+
+	// TODO:
+	// Replace the placeholder connection string below with your
+	// Atlas cluster specifics. Be sure it includes
+	// a valid username and password! Note that in a production environment,
+	// you do not want to store your password in plain-text here.
+	var mongoUri = "<Your Atlas Connection String>"
+
+	// CONNECT TO YOUR ATLAS CLUSTER:
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(
+		mongoUri,
+	))
+
+	defer func() {
+		if err = client.Disconnect(ctx); err != nil {
+			panic(err)
+		}
+	}()
+
+	err = client.Ping(ctx, nil)
+
+	if err != nil {
+		fmt.Println("There was a problem connecting to your Atlas cluster. Check that the URI includes a valid username and password, and that your IP address has been added to the access list. Error: ")
+		panic(err)
+	}
+
+	fmt.Println("Connected to MongoDB!")
+
+	// Provide the name of the database and collection you want to use.
+	// If they don't already exist, the driver and Atlas will create them
+	// automatically when you first write data.
+	var dbName = "myDatabase"
+	var collectionName = "recipes"
+	collection := client.Database(dbName).Collection(collectionName)
+
+	/*      *** INSERT DOCUMENTS ***
+	 *
+	 * You can insert individual documents using collection.Insert().
+	 * In this example, we're going to create 4 documents and then
+	 * insert them all in one call with InsertMany().
+	 */
+
+	eloteRecipe := Recipe{
+		Name:              "elote",
+		Ingredients:       []string{"corn", "mayonnaise", "cotija cheese", "sour cream", "lime"},
+		PrepTimeInMinutes: 35,
+	}
+
+	locoMocoRecipe := Recipe{
+		Name:              "loco moco",
+		Ingredients:       []string{"ground beef", "butter", "onion", "egg", "bread bun", "mushrooms"},
+		PrepTimeInMinutes: 54,
+	}
+
+	patatasBravasRecipe := Recipe{
+		Name:              "patatas bravas",
+		Ingredients:       []string{"potato", "tomato", "olive oil", "onion", "garlic", "paprika"},
+		PrepTimeInMinutes: 80,
+	}
+
+	friedRiceRecipe := Recipe{
+		Name:              "fried rice",
+		Ingredients:       []string{"rice", "soy sauce", "egg", "onion", "pea", "carrot", "sesame oil"},
+		PrepTimeInMinutes: 40,
+	}
+
+	// Create an interface of all the created recipes
+	recipes := []interface{}{eloteRecipe, locoMocoRecipe, patatasBravasRecipe, friedRiceRecipe}
+	insertManyResult, err := collection.InsertMany(context.TODO(), recipes)
+	if err != nil {
+		fmt.Println("Something went wrong trying to insert the new documents:")
+		panic(err)
+	}
+
+	fmt.Println(len(insertManyResult.InsertedIDs), "documents successfully inserted.")
```

Comment:
Nitpick: I know John and Jesse will probably ask for an extra linebreak between every step similar to what they asked me to do for the Node app. E.g.

```
Connected to MongoDB!

4 documents successfully inserted.

elote has 5 ingredients, and takes 35 minutes to make.
fried rice has 7 ingredients, and takes 40 minutes to make.

Found a document with the name fried rice {patatas bravas [potato tomato olive oil onion garlic paprika] 72}

The following document has been updated:
 {
	"Name": "patatas bravas",
	"Ingredients": [
		"potato",
		"tomato",
		"olive oil",
		"onion",
		"garlic",
		"paprika"
	],
	"prepTimeInMinutes": 72
}

Deleted 2 documents in the recipes collection

Reply:
👍